packages/compiler-core/src/transform.ts

```diff
@@ -496,6 +501,15 @@ export function createStructuralDirectiveTransform(
       // structural directive transforms are not concerned with slots
       // as they are handled separately in vSlot.ts
       if (node.tagType === ElementTypes.TEMPLATE && props.some(isVSlot)) {
+        if (
```

Comment:
Look at the comments on lines 501 and 502. 
Maybe move the warn to vSlot.ts?

Reply:
right.. I moved it there and added an unit test for my initial test from #11521 where vue was giving a type error. Now this type error would also happen in the unit tests because the default `onError` for the tests doesn't throw, I've changed the behavior for that one test to throw an error now and make an assertion against that. I haven't seen this done elsewhere so I wondering if you prefer that or rather prefer the code to exit cleanly without any "normal" js errors, though I am a bit unsure where to fix this tbh.